src/maker.rs

```diff
@@ -55,28 +51,26 @@ impl Maker {
             dai_reserved_funds: Default::default(),
             btc_max_sell_amount,
             dai_max_sell_amount,
-            mid_market_rate,
+            mid_market_rate: Some(mid_market_rate),
             spread,
             ongoing_takers: Default::default(),
-            max_rate_ticker_interval_secs
         }
     }
 
     pub fn update_rate(
         &mut self,
         mid_market_rate: MidMarketRate,
     ) -> anyhow::Result<RateUpdateDecision> {
-        // if rate did not change don't create new orders
-        if self.mid_market_rate.value == mid_market_rate.value {
-            // if value did not change update to have newest timestamp
-            self.mid_market_rate = mid_market_rate;
-            return Ok(RateUpdateDecision::RateUpdated);
+        if let Some(previous_mid_market_rate) = self.mid_market_rate {
```

Comment:
:upside_down_face: It may be clearer if you have a single match statement and use match guards.

Reply:
I did it like this because it is an early return for a particular `Some` case - the other `Some` case will be treated like the `None` case.
I think if I don't do it like this I would have to duplicate the logic after the `if let` - but maybe I'm missing something :)

Reply:
I have pushed https://github.com/coblox/nectar/pull/38/commits/f9eab8fec9af4d3599d1641ae7d8a667f2e34ded onto this branch to show you what I meant. See if you like it :D